tests/index_routines.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2021 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# import numpy as np
+
+import cunumeric as num
+
+
```

Comment:
Once all the other issues are solved, could you add a case here testing all possible array dimensionalities on the `a` and `choices` arrays, up to the maximum number of dimensions that Legate was compiled for (`legate.core.LEGATE_MAX_DIM`)?

You will likely want to programmatically generate your inputs. See how I did it in https://github.com/nv-legate/cunumeric/blob/branch-22.01/tests/intra_array_copy.py#L140-L142.

I posted some more input generators that you may want to use on branch-22.01 in `tests/test_tools/generators.py`. From those, `broadcasts_to` will allow you to also test all cases of broadcasting, using code like this:

```
shape = (3,4)
for (np_arr, cn_arr) in zip( broadcasts_to(np, shape, foo), broadcasts_to(cn, shape, foo) ):
  # compare np and cn results
```

This will call `foo(np/cn, shape')` for any `shape'` that broadcasts to `(3,4)`, i.e. `(1,1)`, `(1,4)` and `(3,1)`.